functions/test/endpoints/Event-Tests/AddEvent.test.ts

```diff
@@ -0,0 +1,390 @@
+        const body = {
+            "name": "test event",
+            "details": "test details",
+            "date": "July 11, 2020",
+            "location": "test location",
+            "points": "5",
+            "house": "test house"
+          }
+        const res: request.Test = factory.post(add_event_func, ADD_EVENT_PATH, body, PRIV_RES)
+        res.end(async function (err, res) {
+            if (err) {
+                done(err)
+            } else {
+                expect(res.status).toBe(201)
+
+                let doc = await db.collection("Events").where('Name','==','test event').limit(1).get()
+                expect(doc.docs[0]).toBeDefined()
+                let data = doc.docs[0].data()
+                expect(data).toBeDefined()
+                expect(data.Name).toBe("test event")
+                expect(data.Details).toBe("test details")
+                // Not the best way to test the date but it works
+                expect(data.Date.nanoseconds).toBe(0)
+                expect(data.Date.seconds).toBe(1594440000)
+                expect(data.Location).toBe("test location")
+                expect(data.Points).toBe(5)
+                expect(data.House).toBe("test house")
+                expect(data.CreatorID).toBe(RHP_ID)
+
+                done()
+            }
+        })
+    })
+
+    // Test permission - external advisor
+    it('Test permission - external advisor', async(done) => {
+        const body = {
+            "name": "test event",
+            "details": "test details",
+            "date": "July 11, 2020",
+            "location": "test location",
+            "points": "5",
+            "house": "test house"
+          }
+        const res: request.Test = factory.post(add_event_func, ADD_EVENT_PATH, body, EA_ID)
+        res.end(async function (err, res) {
+            if (err) {
+                done(err)
+            } else {
+                expect(res.status).toBe(201)
+
+                let doc = await db.collection("Events").where('Name','==','test event').limit(1).get()
+                expect(doc.docs[0]).toBeDefined()
+                let data = doc.docs[0].data()
+                expect(data).toBeDefined()
+                expect(data.Name).toBe("test event")
+                expect(data.Details).toBe("test details")
+                // Not the best way to test the date but it works
+                expect(data.Date.nanoseconds).toBe(0)
+                expect(data.Date.seconds).toBe(1594440000)
+                expect(data.Location).toBe("test location")
+                expect(data.Points).toBe(5)
+                expect(data.House).toBe("test house")
+                expect(data.CreatorID).toBe(RHP_ID)
+
+                done()
+            }
+        })
+    })
+
+    // After all of the tests are done, make sure to delete the test firestore app
+    afterAll( () => {
+        Promise.all(firebase.apps().map(app => app.delete()))
+    })
```

Comment:
Add test for competition disabled and competition hidden.

For the concept of POSTing a new instance of a model using the API, it is helpful to create unit tests to handle checks on validity of input data (date range, integer ranges, string lengths). This way we don't have to perform checks for this in our integration tests. 

I know I told you not to worry about unit tests for now, but if you get the chance, I think it would be helpful to add unit tests for the add event function.